pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        properties if not localonly and filters the class based on propertylist
+        and includeClassOrigin.
+
+        Parameters:
+
+          classname (:term:`string`):
+            Name of class to retrieve
+
+          namespace (:term:`string`):
+            Namespace from which to retrieve the class
+
+          params(Dictionary of keywords that determine filtering):
+            The keywords in this dictionary determine any filtering on the
+            class before being returned including LocalOnly, PropertyList,
+            IncludeQualifiers, and IncludeClassOrigin.
+
+        Returns:
+            class if found with superclass properties installed and
+            filtered per the keywords in params.
+        """
+        self._validate_classes_namespace(namespace)
+        try:
+            cc = copy.deepcopy(self.classes[namespace][classname])
+        except KeyError:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Class %s not found' % classname)
+
+        if 'LocalOnly' in params and not params['LocalOnly']:
+            if cc.superclass:
+                try:
+                    del params['ClassName']
+                except KeyError:
+                    pass
+                super_ = self._get_class(cc.superclass, namespace, **params)
+                for prop in super_.properties.values():
+                    if prop.name not in cc.properties:
+                        cc.properties[prop.name] = prop
+                for meth in super_.methods.values():
+                    if meth.name not in cc.methods:
+                        cc.methods[meth.name] = meth
+
+        if 'PropertyList' in params:
+            self._filter_properties(cc, params['PropertyList'])
+
+        if not params['IncludeQualifiers']:
+            self._remove_qualifiers(cc)
+
+        if not params['IncludeClassOrigin']:
+            self._remove_classorigin(cc)
+
+        return cc
+
+    def _get_instance(self, iname, namespace, **params):
+        """
+        Local method implements getinstance. This is generally used by
+        other instance methods that need to get an instance from the
+        repository.
+
+        It attempts to get the instance, copies it, and filters it
+        for input parameters like localonly, includequalifiers, and
+        propertylist.
+        """
+        try:
+            insts = self.instances[namespace]
+        except KeyError:
+            ce = CIMError(CIM_ERR_NOT_FOUND,
+                          'Namespace %s not found' % namespace)
+            raise ce
+
+        rtn_inst = None
+        for inst in insts:
+            if iname == inst.path:
+                if rtn_inst is not None:
+                    raise CIMError(CIM_ERR_FAILED, 'Invalid Repository. '
+                                   'Multiple instances with same path %s'
+                                   % rtn_inst.path)
+                rtn_inst = (copy.deepcopy(inst))
```

Comment:
`CIMinstance` has a `copy()` method. I suggest using that instead of `copy.deepcopy()`.

Reply:
Done